rustup-init.sh

```diff
@@ -705,8 +719,8 @@ check_help_for() {
                     ;;
                 *)
                     # Unknown product version, warn and continue
-                    echo "warn: Detected unknown macOS major version: $(sw_vers -productVersion)"
-                    echo "Warning TLS capabilities detection may fail"
+                    warn "warn: Detected unknown macOS major version: $(sw_vers -productVersion)"
+                    warn "Warning TLS capabilities detection may fail"
```

Comment:
Pre-existing issue, but while we're at it, maybe we can unify all `printf`/`echo` calls related to warnings here by replacing them with `warn`, and the `Warning:`/`Warning` prefixes with `warn:`?

Reply:
Yeah no problem 👌

Reply:
> Might as well do it at the warn function level 🙃
@vic1707 That's okay as well, as long as you make sure the `say` calls aren't affected...